sky/spot/recovery_strategy.py

```diff
@@ -305,6 +306,13 @@ def recover(self) -> float:
                 new_resources = resources.copy(cloud=launched_cloud,
                                                region=launched_region)
                 task.set_resources({new_resources})
+
+                # Clean up preempted TPU VM before launching the cluster.
+                # This is needed as status -r will not remove it if GCP
+                # turns the VM state to other than PREEMPTED.
+                is_tpuvm = tpu_utils.is_tpu_vm(new_resources)
+                if is_tpuvm:
+                    self.terminate_cluster()
```

Comment:
After discussing offline, we found it might be better to have this special case handling in the controller process instead of the recovery strategy. Therefore, future recovery strategies will not need to handle separately.